src/libponyrt/sched/start.c

```diff
@@ -150,6 +165,21 @@ PONY_API int pony_init(int argc, char** argv)
 
   ponyint_cpu_init();
 
+  if (opt.threads == 0) {
+    opt.threads = ponyint_cpu_count();
+  }
+  else if (opt.threads > ponyint_cpu_count())
+  {
```

Comment:
small nit. this can be fixed in another PR. this `{` should be on the preceding line.